3604/1/data_initializer.rb

```diff
@@ -0,0 +1,48 @@
+# class for initialization parametrs, for open file with names of gems and for..
+class DataInitializer
+  # create object for InvalidArgumentError error
+  InvalidArgumentError = Class.new(StandardError)
+
+  attr_reader :argv
+  attr_reader :hash_argv, :array_name_gems
+
+  def initialize
+    @argv = ARGV
+
+    @hash_argv = {
+      top: nil,
+      name: nil,
+      file: 'gems.yml'
+    }
+
+    @array_name_gems = []
```

Comment:
the whole initializer is not needed here. Try to write ur code without it.

Reply:
I do not know how to do it. I can remove argv and use ARGV, but the mentor specifically did do this for me. If I’m putting them in methods, then reek swears: TooManyStatements.